### ISSUE TITLE
Perturbation to find influential features in training data

### ISSUE THREAD

**Issue by vsai121** [kohpangwei/influence-release#11]
Hi,

I am trying to find the influential features of a training instance which affect the test prediction on the MNIST dataset.

This is described in Section 2.2 "Perturbing a training input" as Ipert,loss(Z, Ztest) [Link](https://arxiv.org/pdf/1703.04730.pdf).The authors further describe that this equation can be used to find the features of Z that are most responsible for the prediction on Ztest. 

In my current, approach I use the sign of Ipert,loss(Z,Ztest) to determine the perturbation value. However, I am unsure of how the perturbation of the image should actually be done. 
I cannot find an explanation for that in the paper.

Thanks in advance.

**Comment by Kunlun-Zhu** [comment]
It might be changing the RGB value of the pixel, a guess

**Comment by vsai121** [comment]
Do you suggest an iterative approach, where the pixel value is updated in the direction of sign of Ipert,loss(Z,Ztest)?

**Comment by Kunlun-Zhu** [comment]
I think it's possible that you could update the pixel value to 0 directly

**Comment by vsai121** [comment]
So pixels with a negative Ipert,loss(Z,Ztest) can be updated to zero directly? Is that what you meant?

Thanks a lot for your help :)

**Comment by tengerye** [comment]
@vsai121 In general you can generate any noise x_noise and produce the perturbation as x_per = clip(x+x_noise). The clip makes sure the values are in the reasonable range. In the adversarial examples or similar fields, we get the x_noise=epsilon * d_loss / d_x, where epsilon is a very small scalar.

**Comment by kohpangwei** [comment]
@tengerye thanks for the response!
